examples/methods/plot_subspace.py

```diff
@@ -0,0 +1,336 @@
+
+x_min, x_max = -2.4, 2.4
+y_min, y_max = -2.4, 2.4
+
+
+figsize = (8, 4)
+figure, axes = plt.subplots(1, 2, figsize=figsize)
+
+cm = plt.cm.RdBu
+colormap = ListedColormap(["#FFA056", "#6C4C7C"])
+ax = axes[0]
+ax.set_title("Source data")
+# Plot the source points:
+ax.scatter(Xs[:, 0], Xs[:, 1], c=ys, cmap=colormap, alpha=0.7, s=[15])
+
+ax.set_xticks(()), ax.set_yticks(())
+ax.set_xlim(x_min, x_max), ax.set_ylim(y_min, y_max)
+
+ax = axes[1]
+
+ax.set_title("Target data")
+# Plot the target points:
+ax.scatter(Xt[:, 0], Xt[:, 1], c=ys, cmap=colormap, alpha=0.1, s=[15])
+ax.scatter(Xt[:, 0], Xt[:, 1], c=yt, cmap=colormap, alpha=0.7, s=[15])
+figure.suptitle("Plot of the dataset", fontsize=16, y=1)
+ax.set_xticks(()), ax.set_yticks(())
+ax.set_xlim(x_min, x_max), ax.set_ylim(y_min, y_max)
+
+# %%
+#     Illustration of the problem with no domain adaptation
+# ------------------------------------------
+#
+# When not using domain adaptation, the classifier won't train on
+# data that is distributed as the target sample domain, it will thus
+# not be performing optimaly.
+
+# We create a dict to store scores:
+scores_dict = {}
+
+
+def create_plots(
+    clf,
+    name="Without DA",
+    suptitle=None,
+):
+    size = 16
+
+    if suptitle is None:
+        suptitle = f"Illustration of the {name} method"
+    figure, axes = plt.subplots(1, 3, figsize=(figsize[0] * 1.5, figsize[1]))
+    ax = axes[2]
+    score = clf.score(Xt, yt)
+    DecisionBoundaryDisplay.from_estimator(
+        clf,
+        Xs,
+        cmap=colormap,
+        alpha=0.1,
+        ax=ax,
+        eps=0.5,
+        response_method="predict",
+    )
+
+    # Plot the target points:
+    ax.scatter(
+        Xt[:, 0],
+        Xt[:, 1],
+        c=yt,
+        cmap=colormap,
+        alpha=0.7,
+        s=[15],
+    )
+
+    ax.set_xticks(()), ax.set_yticks(())
+    ax.set_xlim(x_min, x_max), ax.set_ylim(y_min, y_max)
+    ax.set_title("Accuracy on target", fontsize=12)
+    ax.text(
+        x_max - 0.3,
+        y_min + 0.3,
+        ("%.2f" % score).lstrip("0"),
+        size=15,
+        horizontalalignment="right",
+    )
+    scores_dict[name] = score
+    if name != "Without DA":
+        keys = list(clf.named_steps.keys())
+        subspace_estimator = clf.named_steps[keys[0]].base_estimator_
+        clf_on_subspace = clf.named_steps[keys[1]].base_estimator_
+
+        ax = axes[0]
+        # Plot the source points:
+        X_subspace = subspace_estimator.adapt(X)
+        ax.scatter(
+            X_subspace,
+            [0] * X.shape[0],
+            c=y,
+            cmap=colormap,
+            alpha=0.5,
+            s=size,
+        )
+
+        ax.set_xticks(()), ax.set_yticks(())
+        ax.set_ylim(-50, 50)
+        ax.set_title("Full dataset projected on the subspace", fontsize=12)
+
+        ax = axes[1]
+        Xt_adapted = subspace_estimator.adapt(Xt)
+        ax.scatter(
+            Xt_adapted,
+            [0] * Xt.shape[0],
```

Comment:
Don't put label color on target for the first plot

Reply:
sorry, I'm not sure I understood what you want me to do:
I should remove the color for label datapoints in the `Full dataset projected on the subspace` plot?
(and thus add a legend to explain this)